### ISSUE TITLE
TestWatchWithRequireLeader:  no leader watch response error

### ISSUE THREAD

**Issue by fanminshi** [coreos/etcd#7284]
Failing on my master branch of hash 6fb99a858598cf001f545e1253fea004f1e0510b,

```
--- FAIL: TestWatchWithRequireLeader (0.83s)
	watch_test.go:705: expected etcdserver: no leader watch response error, got {Header:{ClusterId:18381416177925296045 MemberId:4790720937027098385 Revision:2 RaftTerm:2} Events:[0xc423b9c1c0] CompactRevision:0 Canceled:false Created:false closeErr:<nil>}
FAIL
exit status 1
```

**Comment by heyitsanthony** [comment]
This is happening because the proxy is polling for leader loss. A WithRequireLeader stream can come in, establish a watcher since `leaderc` won't close until the next leader loss error, receive an event, and then close out on leader loss once the poll cycle completes. The leadership logic will need more thought.

**Comment by heyitsanthony** [comment]
Fixed by #7314 and #7345
